lib/inc/hocon/config_parse_options.hpp

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include <string>
+#include <memory>
+#include "config_syntax.hpp"
+
+namespace hocon {
+    class config_includer;
```

Comment:
Out of curiosity, what's the benefit of this as opposed to just including the config_includer header? Are there cases where one should be preferred to the other?